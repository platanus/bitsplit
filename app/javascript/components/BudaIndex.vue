<template>
  <div id="budaSignInForm">
    <center>
      <div class="w-full max-w-xs">
        <div>
          <textField textvalue1="¡Ya estás conectado a una cuenta Buda! ¿Qué desear hacer?" />
        </div>
        <div class=inline>
          <div class="pb-6">
            <submitButton @do-click="change('BudaForm')" 
            wide="full" color="secondary" :fieldDisabled="false" fieldPlaceholder="Actualizar Keys" />
          </div>
          <div class="pb-6">
            <submitButton @do-click="change('BudaLogoutForm')" 
            wide="full" color="secondary" :fieldDisabled="false" fieldPlaceholder="Desconectar Buda" />
          </div>
        </div>
      </div>
    </center>
  </div>
</template>

<script>
import { mapActions, mapState, mapMutations } from 'vuex'
import textInput from '../components/Input.vue'
import passwordInput from '../components/PasswordInput'
import submitButton from '../components/SubmitButton'
import inputLabel from '../components/InputLabel'
import textField from '../components/TextField'

export default {
  name: 'budaIndex',
  data() {
    return {
      routeName: 'budaIndex',
      api_key: '',
      api_secret: '',
      password: '',
    }
  },
  components: {
    textInput,
    passwordInput,
    submitButton,
    inputLabel,
    textField
  },
  methods: {
    ...mapActions('component', ['changeProfileComp']),
    change(form) {
      this.changeProfileComp(form)
    }
  }
}
</script>

<style scoped>
#budaSignInForm {
  margin-top: 60px;
  margin-bottom: 60px;
}
</style>