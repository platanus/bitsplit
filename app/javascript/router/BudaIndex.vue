<template>
  <div class="mt-16">
    <div>
      <center>
        <div class="w-full max-w-xs">
          <component :is="budaComp" />
        </div>
      </center>
    </div>
  </div>
</template>

<script>
import { mapActions, mapState, mapGetters } from 'vuex';
import BudaForm from '../components/buda/BudaForm';
import BudaLogoutForm from '../components/buda/BudaLogoutForm';
import BudaIndex from '../components/buda/BudaIndex';

export default {
  name: 'BudaProfile',
  data() {
    return {
      routeName: 'budaProfile',
    };
  },
  components: {
    BudaForm,
    BudaLogoutForm,
    BudaIndex,
  },
  computed: {
    ...mapState('component', ['budaComp']),
  },
  methods: {
    ...mapActions('component', ['changeBudaComp']),
    ...mapGetters('user', ['budaSignedIn']),
  },
  created() {
    if (this.budaSignedIn()) {
      this.changeBudaComp('BudaIndex');
    } else {
      this.changeBudaComp('BudaForm');
    }
  },
};
</script>
