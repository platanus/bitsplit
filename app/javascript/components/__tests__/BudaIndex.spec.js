import { shallowMount } from '@vue/test-utils'
import BudaIndex from '../BudaIndex'

describe('BudaIndex', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(BudaIndex)
    expect(wrapper).toMatchSnapshot()
  })
})
