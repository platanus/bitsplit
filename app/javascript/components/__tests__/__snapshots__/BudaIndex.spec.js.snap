// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudaIndex matches snapshot 1`] = `
<div id="budaSignInForm">
  <center>
    <div class="w-full max-w-xs">
      <div>
        <textfield-stub textvalue1="¡Ya estás conectado a una cuenta Buda! ¿Qué desear hacer?"></textfield-stub>
      </div>
      <div class="inline">
        <div class="pb-6">
          <submitbutton-stub color="secondary" wide="full">Actualizar Keys</submitbutton-stub>
        </div>
        <div class="pb-6">
          <submitbutton-stub color="secondary" wide="full">Desconectar Buda</submitbutton-stub>
        </div>
      </div>
    </div>
  </center>
</div>
`;
